pkg/controller/account/account_controller.go

```diff
@@ -254,134 +268,261 @@ func (r *ReconcileAccount) Reconcile(request reconcile.Request) (reconcile.Resul
 		return reconcile.Result{}, err
 	}
 
-	// Fetch the Account instance
-	currentAcctInstance := &awsv1alpha1.Account{}
-	err = r.Client.Get(context.TODO(), request.NamespacedName, currentAcctInstance)
-	if err != nil {
-		if k8serr.IsNotFound(err) {
-			return reconcile.Result{}, nil
+	// If the account is BYOC, needs some different set up
+	if currentAcctInstance.Spec.BYOC {
+
+		reqLogger.Info("BYOC account")
+
+		//Set awsAccountID
+		awsAccountID = currentAcctInstance.Spec.AwsAccountID
+
+		// Get associated AccountClaim
+		accountClaim := &awsv1alpha1.AccountClaim{}
+
+		err := r.Client.Get(context.TODO(),
+			types.NamespacedName{Name: currentAcctInstance.Spec.ClaimLink, Namespace: currentAcctInstance.Spec.ClaimLinkNamespace},
+			accountClaim)
+		if err != nil {
+			if k8serr.IsNotFound(err) {
+				return reconcile.Result{}, nil
+			}
+			return reconcile.Result{}, err
+		}
+
+		// Get credentials
+		byocAWSClient, err := awsclient.GetAWSClient(r.Client, awsclient.NewAwsClientInput{
+			SecretName: accountClaim.Spec.BYOCSecretRef.Name,
+			NameSpace:  accountClaim.Spec.BYOCSecretRef.Namespace,
+			AwsRegion:  "us-east-1",
+		})
+
+		reqLogger.Info("Marking BYOC account claimed")
+		// Ensure the account is marked as claimed
+		if currentAcctInstance.Status.Claimed != true {
+			currentAcctInstance.Status.Claimed = true
+			//return reconcile.Result{}, r.statusUpdate(reqLogger, currentAcctInstance)
```

Comment:
Should we be returning here? What happens if the controller is interrupted here and we recover?

Reply:
Ah yes, I was having problems with the status update here. Should be uncommented.